tests/main/apparmor-prompting-support/task.yaml

```
@@ -0,0 +1,45 @@
+summary: Check that AppArmor Prompting works end-to-end
+
+details: |
+    Test that the AppArmor Prompting subsystems work by simulating common usage
+    scenarios. Manually carry out operations which generate prompts and, using
+    the prompting-client snap in scripted mode, check that those prompts match
+    what is expected, and that the client carries out the operations as intended.
+
+    Details about the operation of the prompting client in scripted mode:
+    https://github.com/canonical/prompting-client/blob/main/docs/running-the-scripted-client.md
+
+
+systems: [- ubuntu-core-*-32*]
+
+prepare: |
+    snap install prompting-client
```

We can do this without an external dependency:
```suggestion
    # Before enabling prompting, we must have a snap installs which provides
    # a prompting `handler-service` with a connection via the
    # `snap-interfaces-requests-control` interface.
    
    # Prerequisite for having a prompts handler service
    snap set system experimental.user-daemons=true
    
    # Install a snap which identifies itself to snapd as a prompting handler-service
    "$TESTSTOOLS"/snaps-state install-local test-snapd-prompt-handler
    snap connect test-snapd-prompt-handler:snap-interfaces-requests-control
```